akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ProtoAnySerialization.scala

```diff
@@ -167,16 +178,52 @@ import scalapb.options.Scalapb
         else
           idAndManifest.substring(0, i).toInt -> idAndManifest.substring(i + 1)
 
+      println(s"# deserialize 3: $typeUrl, akka $id $manifest") // FIXME
+
       serialization.deserialize(scalaPbAny.value.toByteArray, id, manifest).get
     } else if (prefer == Prefer.Scala) {
+      println(s"# deserialize 4: $typeUrl") // FIXME
       // when custom typeUrl
       scalaPbAny
     } else {
+      println(s"# deserialize 5: $typeUrl") // FIXME
       // when custom typeUrl
       ScalaPbAny.toJavaProto(scalaPbAny)
     }
   }
 
+  def toSerializedEvent(scalaPbAny: ScalaPbAny): SerializedEvent = {
+    val typeUrl = scalaPbAny.typeUrl
+    if (typeUrl == ProtoAnyTypeUrl) {
+      // FIXME not sure this is correct
+      val manifest =
+        if (prefer == Prefer.Scala) classOf[ScalaPbAny].getName
+        else classOf[JavaPbAny].getName
+      SerializedEvent(scalaPbAny.toByteArray, akkaProtobufSerializerSerializerId, manifest)
+    } else if (typeUrl.startsWith(GoogleTypeUrlPrefix)) {
+      val manifest = resolveTypeUrl(typeUrl).messageClassName
```

Comment:
I was just thinking about the scenario where producer emits a protobuf message type unknown to the consumer, and where that should break things, should it be here, or should it be in the projection reading the pushed events (to let the producer move on with pushing and not end up with a long backlog)?